geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala

```diff
@@ -61,10 +62,37 @@ trait KafkaDataStoreSchemaManager extends DataStore  {
       AdminUtils.createTopic(zkClient, kfc.topic, partitions, replication)
     }
 
+    // Ensure the topic is ready before adding the new layer to the SFT cache
+    waitUntilTopicReady(zkClient, kfc.topic, partitions)
+
     // put it in the cache
     schemaCache.put(typeName, kfc)
   }
 
+  // Wait until a topic is ready for writes.
+  // NB: The assumption here is that waiting for the leader to be elected is sufficient.
+  // Further, this function delegates to a function which waits for a leader for each partition.
+  def waitUntilTopicReady(zkClient: ZkClient, topic: String, partitions: Int, timeToWait: Long = 5000L): Unit = {
```

Comment:
What is the plan for testing this?  It seems that you need to test
- The looping/sleep/poll logic in a unit test

Would need to wrap the static `ZkUtils.getLeaderForPartition` call in a class and use mockito to mock cases where the method returns normally, never returns, returns late, etc.
- Whether this actually works against zookeeper

This one is harder.  This would be an integration test using zookeeper, perhaps just confirms that it causes no errors.  Ideally, we would want to validate the assumptions, e.g. : 
- Finished leader election is sufficient
- The interactions don't cause other side effects